bytecomp/bytegen.ml

```diff
@@ -1121,29 +1148,25 @@ let comp_remainder cont =
 
 (**** Compilation of a lambda phrase ****)
 
-let compile_implementation modulename expr =
-  Stack.clear functions_to_compile;
+let reset () =
   label_counter := 0;
-  sz_static_raises := [] ;
+  compunit_name := "";
+  Stack.clear functions_to_compile
+
+let compile_implementation modulename expr =
+  reset ();
```

Comment:
Is this `reset` needed given the `~finally` clause below?

Reply:
Not strictly necessary, but it seems to me it helps understanding the logic a bit.
To me, `reset` in `~finally` is about not keeping values alive for no reason. while the `reset` at the beginning is about initializing values correctly before we start using them.